lib/pushmi_pullyu/aip/creator.rb

```diff
@@ -0,0 +1,66 @@
+require 'archive/tar/minitar'
+require 'bagit'
+require 'pushmi_pullyu/aip'
+
+class PushmiPullyu::AIP::Creator
+
+  attr_reader :aip_filename, :aip_directory
+
+  def initialize(noid)
+    @noid = noid
+    @aip_directory = PushmiPullyu::AIP.aip_directory(noid)
+    @aip_filename = PushmiPullyu::AIP.aip_filename(noid)
+    PushmiPullyu::AIP.validate(noid)
+  end
+
+  def run(should_skip_download: false, should_clean_work_directories: true)
+    download_aip unless should_skip_download
```

Comment:
Thanks chris! Looking pretty good. Probably give it a final review and merge it in today